### PR TITLE
CORE-44969 Changing setCustomerIds command name and options.

### PR DESCRIPTION
**Pull request by Aaronius**
## Description
Old API:

```
alloy("setCustomerIds", {
  // IDs here
});
```

New API:

```
alloy("syncIdentity", {
  identities: {
    // IDs here
  }
});
```

## Related Issue
https://jira.corp.adobe.com/browse/CORE-44969

## Motivation and Context
Request by PM as well as flexibility to add options later.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: src/components/Identity/userIds/createUserIds.js [hunk 1]
```diff
@@ -69,5 +69,5 @@ export default ({
     }
   };
 
-  return customerIds;
+  return userIds;
```
Comment (Aaronius) [inline src/components/Identity/userIds/createUserIds.js:72]: I personally find that having this module named `customerIds`/`userIds` is confusing, but I didn't want to get caught up in debates so I didn't attempt to change the pattern for this PR.

File: src/components/Identity/createComponent.js [hunk 2]
```diff
@@ -36,9 +36,9 @@ export default ({
       }
     },
     commands: {
-      setCustomerIds: {
+      syncIdentity: {
         run: options => {
-          return customerIds.sync(options);
+          return userIds.sync(options.userIds);
```
Comment (jonsnyder) [inline src/components/Identity/createComponent.js:41]: We should add some validation so we don't get an error if the user passes the wrong key.
```
objectOf({userIds: objectOf({}).required()}).noUnknownFields().required()
```

**Comment by Aaronius** [comment]
I've added an options validator along with unit tests. We probably should discuss whether/how we'd like to test options validators in the e2e tests across all commands. I've added a JIRA ticket for making sure we have validation set up for all commands and to try to standardize our testing approach: https://jira.corp.adobe.com/browse/CORE-44974

It sounds like since the time this PR was created, there's some hesitancy from PM around whether we'll be using `userIds` or `customerIds`. I'll keep this open a little longer until we either need to merge or receive final word.

**Comment by Aaronius** [comment]
As of a few minutes ago, this:

```
alloy("syncIdentity", {
  userIds: {
    // IDs here
  }
});
```

will be changing to this:

```
alloy("syncIdentity", {
  identities: {
    // IDs here
  }
});
```

I'll try to make the changes tonight.

**Pull request merged**